woocommerce/woocommerce-gutenberg-products-block
# Show compatibility notice when adding the Cart and Checkouts blocks in the editor

Fixes: #3017.
Supersedes: #3018.

Related PR in Gutenberg that intends to remove the page control from the `Guide` when there is only one page: https://github.com/WordPress/gutenberg/pull/29629.

Some notes about this PR:
* The notice is disabled in WP 5.4, in that version of WordPress the `Guide` component we use had a different design, which would require us to have the image in a different size, and also the props were different. Alternatively, we could load it from bundled `wordpress-components`, but that would increase the assets size. Given that we will be dropping support for WP 5.4 in the next weeks, I considered it was not worth it.
* The notice is shown per-user, so each user will see the notice the first time they add the Cart or Checkout blocks or edit a page that contains them. The idea is to behave similarly to the _Welcome guide_ that appears the first time a user opens the block editor.
* The notice needs to be dismissed for each block. Ie: if the user adds the Cart block and dismisses the notice, they will still see a notice when adding the Checkout block.
* My initial idea was to create a store that saves the values into the local storage, similar to `core/edit-post` in Gutenberg. But it seemed a bit overwhelming considering the goal is to save one single value. Instead, I went ahead with a simpler implementation using `useLocalStorageState()`.

### Screenshots
<img src="https://user-images.githubusercontent.com/3616980/110348413-ea71a000-8031-11eb-8103-c343977840dd.png" width="357" alt="Cart and Checkout compatibility notice" />

### How to test the changes in this Pull Request:

1. Create a post or page and add the Cart and Checkout blocks.
2. Verify a compatibility notice appears. Press the <kbd>Tab</kbd> key to verify the focus is inside the modal.
3. Verify the compatibility notice can be dismissed and the block is focused.
4. Verify reloading the page or adding the same block in another page or post doesn't show the notice again.
